### PR TITLE
Mgur27/use crypto pelion

### PR DESCRIPTION
**Pull request by micgur01**
update mbed-os 
use mbedtls crypto pelion and updated mbedtls config to reduce size
enabled encryption for all target with external memory
update fota header size from 0x400 to 0x200
use armc6 to reduce code size for disco and nucleo 411 targets

still not final version as synced not from mcc nanoclient

### CODE REVIEW

File: source/bootloader_mbedtls_user_config.h [hunk 1]
```diff
@@ -1,59 +1,196 @@
...
-// Reduces ROM size by 30 kB
-#undef MBEDTLS_ERROR_STRERROR_DUMMY
-#undef MBEDTLS_VERSION_FEATURES
-#undef MBEDTLS_DEBUG_C
-
-#include "mbedtls/check_config.h"
-
-#endif /* MBEDTLS_CUSTOM_CONFIG_H */
+// ----------------------------------------------------------------------------
+// Copyright 2016-2017 Peloion Ltd.
```
Comment (jenia81) [inline source/bootloader_mbedtls_user_config.h]: please fix the license and also the file access was changed

Comment (micgur01) [reply source/bootloader_mbedtls_user_config.h]: done
file access - good change

File: presets/build_presets.py [hunk 2]
```diff
@@ -65,8 +64,12 @@ def main():
             continue
         for preset in target.iterdir():
             app_config = preset.joinpath('mbed_app.json').relative_to(root_dir)
-            cmd = BUILD_CMD_TEMPLATE.format(
-                target=target_name, config=app_config.as_posix()).split(' ')
+            if (target_name == "NUCLEO_F411RE" or target_name == "DISCO_L475VG_IOT01A"):
```
Comment (jenia81) [inline presets/build_presets.py:68]: please add a comment why those 2 targets are compiled with ARM

File: source/fix-compile/armc6.c [hunk 3]
```diff
@@ -1,5 +1,5 @@
 // ----------------------------------------------------------------------------
-// Copyright 2019-2021 Pelion Ltd.
+// Copyright 2019-2021 Peloion Ltd.
```
Comment (jenia81) [inline source/fix-compile/armc6.c]: license

File: source/fix-compile/armc6.c [hunk 4]
```diff
@@ -16,18 +16,19 @@
...
-#define MBEDTLS_ASN1_ENUMERATED              0x0A
+#include "fota/fota_crypto.h"
 
-int mbedtls_asn1_get_enumerated_value(unsigned char **p,
-                                      const unsigned char *end,
-                                      int *val);
-int mbedtls_asn1_get_int64(unsigned char **p,
-                           const unsigned char *end,
-                           int64_t *val);
+int fota_verify_signature_prehashed(
```
Comment (jenia81) [inline source/fix-compile/armc6.c:28]: Please add a comment here why this stub is needed

File: source/mbed-trace/mbed_trace.h [hunk 5]
```diff
@@ -1,5 +1,5 @@
 // ----------------------------------------------------------------------------
-// Copyright 2018 ARM Ltd.
+// Copyright 2018 Peloion Ltd.
```
Comment (jenia81) [inline source/mbed-trace/mbed_trace.h]: license

File: source/storage/fota_device_key.cpp [hunk 6]
```diff
@@ -1,5 +1,5 @@
 // ----------------------------------------------------------------------------
-// Copyright 2021 ARM Ltd.
+// Copyright 2021 Peloion Ltd.
```
Comment (jenia81) [inline source/storage/fota_device_key.cpp]: license

File: source/storage/fota_platform_default_bd.cpp [hunk 7]
```diff
@@ -1,5 +1,5 @@
 // ----------------------------------------------------------------------------
-// Copyright 2019 ARM Ltd.
+// Copyright 2019 Peloion Ltd.
```
Comment (jenia81) [inline source/storage/fota_platform_default_bd.cpp]: license

File: storage/include/CloudClientStorage.h [hunk 8]
```diff
@@ -1,5 +1,5 @@
 // ----------------------------------------------------------------------------
-// Copyright 2016-2019 ARM Ltd.
+// Copyright 2016-2019 Peloion Ltd.
```
Comment (jenia81) [inline storage/include/CloudClientStorage.h]: license

File: upgrade.c [hunk 9]
```diff
@@ -1,5 +1,5 @@
 // ----------------------------------------------------------------------------
-// Copyright 2019-2021 ARM Ltd.
+// Copyright 2019-2021 Peloion Ltd.
```
Comment (jenia81) [inline upgrade.c]: license

**Comment by micgur01** [comment]
push the code review fixes

**Review by jenia81 — APPROVED** [review]

**Pull request merged**